Files acpitool-0.4.7.old/src/acpitool and acpitool-0.4.7/src/acpitool differ
diff -Nru acpitool-0.4.7.old/src/acpitool.cpp acpitool-0.4.7/src/acpitool.cpp
--- acpitool-0.4.7.old/src/acpitool.cpp	2007-04-29 14:38:46.289190064 +0000
+++ acpitool-0.4.7/src/acpitool.cpp	2007-04-29 14:43:59.411033861 +0000
@@ -54,15 +54,15 @@
     char *filename;
     
     filename = "/proc/acpi/info";
-    
     file_in.open(filename);
-    if (!file_in)
-    	return 0;
-    else
-    {
-	file_in.close();
-        return 1;
+    if(!file_in.is_open()) {
+        filename = "/sys/module/acpi/parameters/acpica_version";
+        file_in.open(filename);
     }
+    if(!file_in.is_open()) return 0;
+
+    file_in.close();
+    return 1;
 }
     
 int Print_ACPI_Info(int show_ac, int show_therm, int show_trip, int show_fan, int show_batteries, int show_empty, int show_version, int show_cpu, int show_wake, int e_set, int info_level, int verbose)
@@ -96,23 +96,24 @@
     char *filename, str[50];
     
     filename = "/proc/acpi/info";
-    
     file_in.open(filename);
-    if (!file_in)
-    {
-    	if(!verbose)
-    	{
-	    strcpy(c, "<n.a>");
-    	    return 0;
-    	}
-    	else
-    	{
-    	    cout<<"  Could not open file : "<<filename<<endl;
-	    cout<<"  Make sure your kernel has ACPI support enabled."<<endl;
-	    return -1;
-    	}	
+    if(!file_in.is_open()) {
+        filename = "/sys/module/acpi/parameters/acpica_version";
+        file_in.open(filename);
+    }
+    if(!file_in.is_open()) {
+	    // failure
+	    if(!verbose) {
+		    strcpy(c, "<n.a>");
+		    return 0;
+	    }
+	    else {
+		    cout<<"  Could not open file : "<<filename<<endl;
+		    cout<<"  Make sure your kernel has ACPI support enabled."<<endl;
+		    return -1;
+	    }
     }
-       
+
     file_in.getline(str, 50);
     file_in.close();
     
Files acpitool-0.4.7.old/src/acpitool.o and acpitool-0.4.7/src/acpitool.o differ
